### PR TITLE
Blood brothers no longer need their entire team to escape, just one

### PR DESCRIPTION
**Pull request by Simplehorror**
# Document the changes in your pull request

In the end, blood brothers are just human tools. And I don't remember losing my emag and then the Syndicate saying I didn't win because I lost my emag or my emag went AFK in a locker.
This also opens up new avenues for blood brothers to explore, such as one suicidebombing brig (which he knows he can still win with)

# Spriting

# Wiki Documentation

Blood brothers no longer need their entire team to escape, just one
# Changelog

:cl:  
tweak: Blood brothers no longer need their entire team to escape, just one
/:cl:

### CODE REVIEW

**Comment by ynot01** [comment]
Should make sure the brother that escapes has all of the objectives completed (steal objs being forefront in my mind)

**Comment by Simplehorror** [comment]
> Should make sure the brother that escapes has all of the objectives completed (steal objs being forefront in my mind)

steal objectives are only accomplished if you're on the shuttle iirc, will take another look

**Comment by Simplehorror** [comment]
for anyone against this PR: please tell me how TC-less traitor, except I have to babysit some 10 hour player who can barely move to get to the shuttle is fun. hijack BB is easier because I don't have to deal with that

**Comment by ReddicusDragon** [comment]
while it may not be fun to you, it may be fun to others. Additionally, it basically removes the purpose of BB. You are supposed to take care of your brother. If you red-text because your brother is a 10 hour player, oh well, next time you're antag maybe you can do something different

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -519,6 +519,19 @@ GLOBAL_LIST_EMPTY(objectives)
...
 
+/datum/objective/escape/no_team_escape
+	team_explanation_text = "Have at least one member of your team escape on a shuttle or pod alive, without being in custody."
+
+/datum/objective/escape/no_team_escape/check_completion()
+	if(..())
+		return TRUE
+	// Require all owners escape safely.
+	var/list/datum/mind/owners = get_owners()
+	for(var/datum/mind/M in owners)
```
Comment (ReddicusDragon) [inline code/game/gamemodes/objective.dm:530]: this loop literally returns this objective as complete if anyone in the owners list is considered escape. This is the epitome of shitcode.

File: code/game/gamemodes/objective.dm [hunk 2]
```diff
@@ -706,7 +721,8 @@ GLOBAL_LIST_EMPTY(possible_items)
 	for(var/datum/mind/M in owners)
 		if(!isliving(M.current))
 			continue
-
+		if (!considered_escaped(M))
```
Comment (ReddicusDragon) [inline code/game/gamemodes/objective.dm:724]: what is this check for? You don't check in this proc if they're considered escaped to begin with

**Review by ReddicusDragon — CHANGES_REQUESTED** [review]
overall a bad pr

File: code/game/gamemodes/objective.dm (hunk 1 above)
Comment (Simplehorror) [reply code/game/gamemodes/objective.dm:530]: yes thats the point

File: code/game/gamemodes/objective.dm (hunk 2 above)
Comment (Simplehorror) [reply code/game/gamemodes/objective.dm:724]: read the commit name

**Comment by Simplehorror** [comment]
> while it may not be fun to you, it may be fun to others. Additionally, it basically removes the purpose of BB. You are supposed to take care of your brother. If you red-text because your brother is a 10 hour player, oh well, next time you're antag maybe you can do something different

i like how you literally admit i'm just fucked if i get a 10 hour player because there is 0 chance that he will be escaping alive. how is that fun?

File: code/game/gamemodes/objective.dm (hunk 2 above)
Comment (ReddicusDragon) [reply code/game/gamemodes/objective.dm:724]: Okay I see what you're doing here

**Comment by ReddicusDragon** [comment]
> i like how you literally admit i'm just fucked if i get a 10 hour player because there is 0 chance that he will be escaping alive. how is that fun?

The point of blood brother is to collaborate. You try your best. Your blood brother dying is outside of your control. Just because you are antag does not mean you will greentext every time, bud.

**Comment by ReddicusDragon** [comment]
Plus BB has communication implants now

**Comment by Simplehorror** [comment]
> > i like how you literally admit i'm just fucked if i get a 10 hour player because there is 0 chance that he will be escaping alive. how is that fun?
> 
> The point of blood brother is to collaborate. You try your best. Your blood brother dying is outside of your control. Just because you are antag does not mean you will greentext every time, bud.

yes it's out of my control so why should I be punished

**Comment by ReddicusDragon** [comment]
> > > i like how you literally admit i'm just fucked if i get a 10 hour player because there is 0 chance that he will be escaping alive. how is that fun?
> > 
> > 
> > The point of blood brother is to collaborate. You try your best. Your blood brother dying is outside of your control. Just because you are antag does not mean you will greentext every time, bud.
> 
> yes it's out of my control so why should I be punished

Your issue is you viewing a blood brother who is new to being blood brother as a punishment. They're a new player, they are learning. If you want them to improve you should be helping them. If you don't try to teach them, they won't improve and will be just as bad as a 200 hr blood brother

**Comment by Therandomhoboo** [comment]
This kinda defeats the point of a BB, the point is teamwork and to leave together. If you leave without your BB then whats the point in having them? Might as well just be Blood Son instead of Blood Brothers at that point

**Comment by Simplehorror** [comment]
blood brothers' point is not teamwork exactly, it just replaces the currency of TC with another guy which can be useful or it can be detrimental. this removes the detrimental part.

**Comment by MajManatee** [comment]
the point of BB is 100% teamwork.
This PR basically just makes it so you are a solo traitor with no TC, and have someone else to talk to about it.
Turn off BB if you dont like working together.

**Comment by Simplehorror** [comment]
> the point of BB is 100% teamwork. This PR basically just makes it so you are a solo traitor with no TC, and have someone else to talk to about it. Turn off BB if you dont like working together.

that's awesome. i'm going to turn on blood brother and if I get it with you I'm going to act completely clueless and randomly die to some environment hazard half way through the round, and we'll see if you turn it off or not

**Comment by Bibby0110** [comment]
Blood brothers is meant to be cooperative. This includes escaping. We’re open to alternative ideas though, keep doing what you’re doing simplehorror :)

**Pull request closed without merging**